pkg/apis/operator/v1alpha1/tektondashboard_lifecycle.go

```diff
@@ -26,102 +26,156 @@ var (
-// message.
-func (tps *TektonDashboardStatus) MarkInstallFailed(msg string) {
-	dashboardCondSet.Manage(tps).MarkFalse(
-		InstallSucceeded,
-		"Error",
-		"Install failed with message: %s", msg)
+func (tds *TektonDashboardStatus) MarkInstallerSetAvailable() {
+	dashboardCondSet.Manage(tds).MarkTrue(InstallerSetAvailable)
 }
 
-// MarkDeploymentsAvailable marks the DeploymentsAvailable status as true.
-func (tps *TektonDashboardStatus) MarkDeploymentsAvailable() {
-	dashboardCondSet.Manage(tps).MarkTrue(DeploymentsAvailable)
+func (tds *TektonDashboardStatus) MarkInstallerSetReady() {
+	dashboardCondSet.Manage(tds).MarkTrue(InstallerSetReady)
 }
 
-// MarkDeploymentsNotReady marks the DeploymentsAvailable status as false and calls out
-// it's waiting for deployments.
-func (tps *TektonDashboardStatus) MarkDeploymentsNotReady() {
-	dashboardCondSet.Manage(tps).MarkFalse(
-		DeploymentsAvailable,
-		"NotReady",
-		"Waiting on deployments")
+func (tds *TektonDashboardStatus) MarkPostReconcilerComplete() {
+	dashboardCondSet.Manage(tds).MarkTrue(PostReconciler)
 }
 
 // MarkDependenciesInstalled marks the DependenciesInstalled status as true.
-func (tps *TektonDashboardStatus) MarkDependenciesInstalled() {
-	dashboardCondSet.Manage(tps).MarkTrue(DependenciesInstalled)
+func (tds *TektonDashboardStatus) MarkDependenciesInstalled() {
+	dashboardCondSet.Manage(tds).MarkTrue(DependenciesInstalled)
+}
+
+func (tds *TektonDashboardStatus) MarkNotReady(msg string) {
+	dashboardCondSet.Manage(tds).MarkFalse(
+		apis.ConditionReady,
+		"Error",
+		"Ready: %s", msg)
+}
+
+func (tds *TektonDashboardStatus) MarkPreReconcilerFailed(msg string) {
+	tds.MarkNotReady("PreReconciliation failed")
+	dashboardCondSet.Manage(tds).MarkFalse(
+		PreReconciler,
+		"Error",
+		"PreReconciliation failed with message: %s", msg)
+}
+
+func (tds *TektonDashboardStatus) MarkInstallerSetNotAvailable(msg string) {
+	tds.MarkNotReady("TektonInstallerSet not ready")
+	dashboardCondSet.Manage(tds).MarkFalse(
+		InstallerSetAvailable,
+		"Error",
+		"Installer set not ready: %s", msg)
+}
+
+func (tds *TektonDashboardStatus) MarkInstallerSetNotReady(msg string) {
+	tds.MarkNotReady("TektonInstallerSet not ready")
+	dashboardCondSet.Manage(tds).MarkFalse(
+		InstallerSetReady,
+		"Error",
+		"Installer set not ready: %s", msg)
+}
+
+func (tds *TektonDashboardStatus) MarkPostReconcilerFailed(msg string) {
+	tds.MarkNotReady("PostReconciliation failed")
+	dashboardCondSet.Manage(tds).MarkFalse(
+		PostReconciler,
+		"Error",
+		"PostReconciliation failed with message: %s", msg)
 }
 
 // MarkDependencyInstalling marks the DependenciesInstalled status as false with the
 // given message.
-func (tps *TektonDashboardStatus) MarkDependencyInstalling(msg string) {
-	dashboardCondSet.Manage(tps).MarkFalse(
+func (tds *TektonDashboardStatus) MarkDependencyInstalling(msg string) {
+	dashboardCondSet.Manage(tds).MarkFalse(
```

Comment:
`tts.MarkNotReady("Dependencies installing")`

Reply:
pls check https://github.com/tektoncd/operator/blob/9b97efa1f5dafb07a88847daa255bd032fcf498d/pkg/apis/operator/v1alpha1/tektontrigger_lifecycle.go

Reply:
we add error in Ready as TektonConfig only watches ready cond